Terminal.Gui/Core/KeyBindings.cs

```diff
@@ -0,0 +1,370 @@
+		/// <param name="outKey">The current output key.</param>
+		/// <param name="toReplace">The enabled state to replace.</param>
+		/// <returns><c>true</c> if enabled status was successfully sets, <c>false</c>otherwise.</returns>
+		public bool EnableDisableKeyBinding (string view, Key inKey, Key outKey, bool toReplace)
+		{
+			if (view == null) {
+				throw new ArgumentNullException ("View cannot be null.", nameof (view));
+			}
+			Type v = GetInstance (view).GetType ();
+			return EnableDisableKeyBinding (v, inKey, outKey, toReplace);
+		}
+
+		/// <summary>
+		/// Enables or disables the <see cref="KeyBinding.Enabled"/> of the match key binding.
+		/// </summary>
+		/// <param name="keyBinding">The key binding.</param>
+		/// <param name="toReplace">The enabled state to replace.</param>
+		/// <returns><c>true</c> if enabled status was successfully sets, <c>false</c>otherwise.</returns>
+		public bool EnableDisableKeyBinding (KeyBinding keyBinding, bool toReplace)
+		{
+			if (keyBinding == null) {
+				throw new ArgumentNullException ("KeyBinding cannot be null.", nameof (keyBinding));
+			}
+			Type view = GetInstance (keyBinding.View).GetType ();
+			return EnableDisableKeyBinding (view, keyBinding.InKey, keyBinding.OutKey, toReplace);
+		}
+
+		/// <summary>
+		/// Replaces the match <see cref="KeyBinding"/> to the new binding sets.
+		/// </summary>
+		/// <param name="from"></param>
+		/// <param name="to"></param>
+		/// <returns><c>true</c> if the key binding was successfully replaced, <c>false</c>otherwise.</returns>
+		public bool ReplaceViewKey (KeyBinding from, KeyBinding to)
+		{
+			if (from == null) {
+				throw new ArgumentNullException ("KeyBinding cannot be null.", nameof (from));
+			}
+			if (to == null) {
+				throw new ArgumentNullException ("KeyBinding cannot be null.", nameof (to));
+			}
+			if (from.View != to.View) {
+				throw new InvalidOperationException ("Both views must be the same.");
+			}
+			if (new KeyBindingEqualityComparer ().ReferenceEquals (from, to)) {
+				throw new ArgumentException ("Both KeyBinding are equal.");
+			}
+			var viewName = GetInstance (from.View).GetType ().Name;
+			var idxFrom = Keys.FindIndex (x => x.View == viewName && x.InKey == from.InKey && x.OutKey == from.OutKey);
+			if (idxFrom == -1) {
+				throw new ArgumentException ("KeyBinding not found.", nameof (idxFrom));
+			}
+			var idxTo = Keys.FindIndex (x => x.View == viewName && (x.InKey == to.InKey || x.OutKey == to.OutKey));
+			if (idxTo > -1 && idxTo != idxFrom) {
+				throw new ArgumentException ("One of the keys already exists.", nameof (idxTo));
+			}
+			var kb = Keys [idxFrom];
+			kb.InKey = to.InKey;
+			kb.OutKey = to.OutKey;
+			kb.Description = to.Description;
+			kb.Enabled = to.Enabled;
+			if (new KeyBindingEqualityComparer().ReferenceEquals (kb, to)) {
+				return true;
+			}
+			return false;
+		}
+
+		private View GetInstance (string strFullyQualifiedName)
```

Comment:
I don't think this method (`View GetInstance(string)`) will work for user defined Views since they will not be in `(View).Assembly`.  

This method seems to only be used to get the `Type` anyway (e.g. `GetInstance (from.View).GetType ().Name`).  Is there a reason not to just do everything at the `string` level instead of with `Type`.  The overloads that take a `Type` could stay but just call `.Name` on those asap and do everything with the class name?

Reply:
As you can see I store the views as string. I would like to pass as string but I have no warranty that the view is based in the `View` class. Imagine that a user derives from a `Dialog` view and how I have to check if the derived class has the base class `View`? Look that the `Dialog` is derived from `Window`, which is derived from `Toplevel`, which is derived from `View`. I don't want allow store a key binding to a view that is not a valid `View` class. What is the better approach?

Reply:
> I don't want allow store a key binding to a view that is not a valid `View` class. [...]

I think it is fair to trust that the API user can provide a `Type` or `nameof(MyViewType)` without checking ourselves that that is definetly derrived from `View`.  The worst case scenario is that they end up with a keybinding that does nothing right? And they should figure out what they have done wrong quite quickly when testing their code (assuming the method xmldoc documents this expectation).